### PR TITLE
Add ability to search for changes made by a user

### LINKED ISSUE #13: Handle missing whodunnits?
User make get deleted,etc/

### LINKED ISSUE #14: Add Journal by whodunnit
Would like to be able to get output in LifeHistory style human readable form of the details filtered by whodunnit over a time range. This would have to add the item being altered to the LifeHistory style output since the context would be a whodunnit individual.

### PR DESCRIPTION
**Pull request by dinjas**
This PR aims to fix #14 by providing the ability to search for changes made by a specific user.
 
Usage for this would be like: `puts PaperTrailScrapbook::UserJournal.new(user).story`

Also fixes #13

### CODE REVIEW

File: lib/paper_trail_scrapbook/version_helpers.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          if whodunnit_class.method_defined?(:to_whodunnit)
+            whodunnit_class.find(author).to_whodunnit
+          else
+            whodunnit_class.find(author).to_s
+          end
+        else
+          author
+        end
+      else
+        config.unknown_whodunnit
```
Comment (tjchambers) [inline lib/paper_trail_scrapbook/version_helpers.rb]: short circuit return - I think this method could be cleaned up

File: spec/paper_trail_scrapbook/user_journal_spec.rb [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    end
+
+    before do
+      PaperTrailScrapbook.config.whodunnit_class = Person
+      PaperTrail.whodunnit = person.id
+    end
+
+    let!(:book) { Book.create!(title: 'How the Grinch stole Xmas') }
+    let(:book2) { Book.create!(title: 'Green Eggs and Ham') }
+    let!(:author) { Person.create!(name: 'Dr. Seuss') }
```
Comment (tjchambers) [inline spec/paper_trail_scrapbook/user_journal_spec.rb]: add the let!s to the before do

File: spec/paper_trail_scrapbook/user_journal_spec.rb [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        expect(subject).to match(/ • title: How the Grinch stole Xmas/)
+        expect(subject).to match(/On .*, created Person\[#{author.id}\]:/)
+        expect(subject).to match(/ • name: Dr. Seuss/)
+        expect(subject).to match(/On .*, created Authorship\[#{target.id}\]:/)
+        expect(subject)
+          .to match(/ • book: How the Grinch stole Xmas\[#{book.id}\]/)
+      end
+
+      it 'provides a whole story missing to_s content' do
+        book.title = nil
```
Comment (tjchambers) [inline spec/paper_trail_scrapbook/user_journal_spec.rb]: update!?

File: spec/paper_trail_scrapbook/user_journal_spec.rb [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    let(:book2) { Book.create!(title: 'Green Eggs and Ham') }
+    let!(:author) { Person.create!(name: 'Dr. Seuss') }
+    let!(:target) { Authorship.create!(book: book, author: author) }
+
+    let(:format) { PaperTrailScrapbook.config.time_format }
+    let(:object) { described_class.new(person, {}) }
+    let(:subject) { object.story }
+
+    describe '#story' do
+      it 'provides a whole story' do
```
Comment (tjchambers) [inline spec/paper_trail_scrapbook/user_journal_spec.rb]: I would introduce some contexts, extract the setup to let's and before/do to structure this entire spec better

**Comment by dinjas** [comment]
Think this is ready for another gander. I'm open to suggestions for addressing code climate recommendations.

**Pull request merged**